tests/test_interpolation.py

```diff
@@ -1099,9 +1099,186 @@ def test_inject_subdomain_sinc(self):
                          'p_sr0rsr0xrsr0y')
 
     @pytest.mark.parallel(mode=4)
+    def test_interpolate_subdomain_mpi_mfe(self, mode):
+        """
+        MFE: Simplified test case for subdomain interpolation bug.
+
+        Regression test for bug where multiple interpolations in one Operator
+        would reuse the same temporary variable name ('sum'), causing garbage
+        values when ConditionalDimensions prevented some interpolations from
+        executing (points outside subdomain boundaries).
+
+        The bug was fixed by making temporary variable names unique per
+        SparseFunction: Symbol(name=f'sum_{self.sfunction.name}', ...)
+
+        This test uses a simpler setup than test_interpolate_subdomain_mpi
+        but still exercises the core bug condition: multiple subdomain
+        interpolations in one Operator with MPI.
+
+        NOTE: Using shape=(41, 41) to avoid degenerate MPI decomposition
+        with 4 ranks. Need at least ~10 points per rank for proper halos.
+        """
+        grid = Grid(shape=(41, 41), extent=(10., 10.))
+
+        # SD2 covers the left 4 points in x dimension: x in [0, 4)
+        sd2 = SD2(grid=grid)
+
+        # Function defined ONLY on the subdomain
+        f0 = Function(name='f0', grid=sd2)
+
+        # Initialize with mesh data
+        xmsh, ymsh = np.meshgrid(np.arange(41), np.arange(41))
+        msh = xmsh*ymsh  # msh[i,j] = i*j
+        f0.data[:] = msh[:20, :]
```

Comment:
This does not match with the earlier comment (or indeed `SD2` definition). Also doesn't address the size of the subdomain at all.